pkg/scheduler/scheduler.go

```diff
@@ -256,6 +256,7 @@ func (s *Scheduler) admit(ctx context.Context, e *entry) error {
 	}
 	newWorkload.Spec.Admission = admission
 	if err := s.cache.AssumeWorkload(newWorkload); err != nil {
```

Comment:
If there is an error, `AssumeWorkload` doesn't store anything. So this change is not doing anything. Unless I'm missing something?

Reply:
Oh, I see. You are trying to remove the information from a previous assumption.

If so, this is not the right place to do this. We shouldn't have a workload indefinitely in assumed state. This should have been cleared when a workload fails to be admitted. Where are we forgetting to clear it?

Reply:
This is the only place where `admit()` will return error. Or we can add here https://github.com/kubernetes-sigs/kueue/blob/4ec686fa7dde59579db3eb6ba699831e3bed18ac/pkg/scheduler/scheduler.go#L123

Reply:
This is why I add the `ForgetWorkload` here, honestly, I haven't figure out why this happen for the first time. But considering when `AssumeWorkload` failed, we'll requeue immediately, so I think it's safe to add reverse operation here.

Reply:
It's not safe. There could be a second scheduling attempt for a workload that it's in the "binding phase". We need to wait for that one to finish.